src/MicrosoftAi/AbstractionMapper.cs

```diff
@@ -0,0 +1,321 @@
+		};
+	}
+
+	/// <summary>
+	/// Converts a collection of Microsoft.Extensions.AI.<see cref="AITool"/> to a collection of OllamaSharp tools.
+	/// </summary>
+	/// <param name="tools">The tools to convert.</param>
+	private static IEnumerable<Tool>? ToOllamaSharpTools(IEnumerable<AITool>? tools)
+	{
+		return tools?.Select(ToOllamaSharpTool)
+					 .Where(t => t is not null)
+					 .Cast<Tool>();
+	}
+
+	/// <summary>
+	/// Converts an Microsoft.Extensions.AI.<see cref="AITool"/> to an OllamaSharp tool.
+	/// </summary>
+	/// <param name="tool">The tool to convert.</param>
+	private static Tool? ToOllamaSharpTool(AITool tool)
+	{
+		if (tool is AIFunction f)
+			return ToOllamaSharpTool(f.Metadata);
+
+		return null;
+	}
+
+	/// <summary>
+	/// Converts <see cref="AIFunctionMetadata"/> to a <see cref="Tool"/>.
+	/// </summary>
+	/// <param name="functionMetadata">The function metadata to convert.</param>
+	private static Tool? ToOllamaSharpTool(AIFunctionMetadata functionMetadata)
+	{
+		return new Tool
+		{
+			Function = new Function
+			{
+				Description = functionMetadata.Description,
+				Name = functionMetadata.Name,
+				Parameters = new Parameters
+				{
+					Properties = functionMetadata.Parameters.ToDictionary(p => p.Name, p => new Properties
+					{
+						Description = p.Description,
+						Enum = [], // TODO is there such as possible values in AIFunctionParameterMetadata?
+						Type = ToFunctionTypeString(p.ParameterType)
+					}),
+					Required = functionMetadata.Parameters.Where(p => p.IsRequired).Select(p => p.Name),
+					Type = "object"
+				}
+			},
+			Type = "function"
+		};
+	}
+
+	/// <summary>
+	/// Converts a <see cref="Type"/> to a function type string.
+	/// </summary>
+	/// <param name="_">The type to convert.</param>
+	private static string ToFunctionTypeString(Type? _)
+	{
+		return "string"; // TODO others supported?
+	}
+
+	/// <summary>
+	/// Converts a list of Microsoft.Extensions.AI.<see cref="ChatMessage"/> to a list of Ollama <see cref="Message"/>.
+	/// </summary>
+	/// <param name="chatMessages">The chat messages to convert.</param>
+	private static IEnumerable<Message> ToOllamaSharpMessages(IList<ChatMessage> chatMessages)
+	{
+		foreach (var cm in chatMessages)
+		{
+			yield return new Message
+			{
+				Content = cm.Text,
+				Images = cm.Contents.OfType<DataContent>().Select(ToOllamaImage).Where(s => !string.IsNullOrEmpty(s)).ToArray(),
+				Role = ToOllamaSharpRole(cm.Role),
+				ToolCalls = cm.Contents.OfType<FunctionCallContent>().Select(ToOllamaSharpToolCall),
+			};
+		}
+	}
+
+	/// <summary>
+	/// Converts a Microsoft.Extensions.AI.<see cref="DataContent"/> to a base64 image string.
+	/// </summary>
+	/// <param name="content">The data content to convert.</param>
+	private static string ToOllamaImage(DataContent content)
+	{
+		if (content is null || !content.ContainsData)
+			return string.Empty;
+
+		if (content.MediaType?.StartsWith("image", StringComparison.OrdinalIgnoreCase) ?? false)
```

Comment:
You might want to also check if `content` is an `ImageContent`. In that case you should supply it to Ollama regardless of whether the media type is specified.